### ISSUE TITLE
Support for enterprise gitHub?

### ISSUE THREAD

**Issue by twastvedt** [neuecc/Open-on-GitHub#18]
Is there a way to get this to work with an enterprise gitHub server? (git.companyName.com) If not supported, then I guess this is a feature request! :) Maybe a configurable domain name?

**Comment by ridilculous** [comment]
Until some (months?) ago i've been successfully using this extension with github enterprise and VS2022. Now the context menu always shows `error:NullReferenceException`.

**Comment by neuecc** [comment]
maybe v2.0.0 changes are affected.
https://github.com/neuecc/Open-on-GitHub/pull/17
I have no plans to implement it, but I will merge it when the PR comes in.

**Comment by ycherkes** [comment]
@ridilculous @twastvedt  Could you give me example of all possible GitHub enterprise URLs? I will try to fix the issue.

**Comment by ridilculous** [comment]
> @ridilculous @twastvedt Could you give me example of all possible GitHub enterprise URLs? I will try to fix the issue.

@ycherkes I don't think i know all possibilities. For me it looks like they're exactly the same as the ones from github.com, except that the domain is different, f..e

```
https://my.org/suborg/proj/tree/branchname
https://my.org/suborg/proj/blob/branchname/filename
https://my.org/suborg/proj/blob/branchname/filename#L89
https://longer.url.of.my.org/anothersuborg/anotherproj/commit/somehash
https://longer.url.of.my.org/anothersuborg/anotherproj/commit/somehash#diff-anotherhash
```

**Comment by twastvedt** [comment]
I agree with @ridilculous. Urls on our server look identical to public github urls once you get to the suborg part of the url.

**Comment by ycherkes** [comment]
@ridilculous @twastvedt If I understood you correctly, there is no way to recognize such kind of url as Github enterprise?
What do you think, it's OK if I interpret the url as Github enterprise if it doesn't match Gitea, Bitbucket or AzureDevOps pattern? Or it's better to create options page with ability to configure custom url pattern and host type (Gitea, Bitbucket, AzureDevOps, Github)?

**Comment by ycherkes** [comment]
@neuecc @ridilculous @twastvedt 
Could you please take a look at [PR#20](https://github.com/neuecc/Open-on-GitHub/pull/20)?
It should fix this issue.

**Comment by neuecc** [comment]
thanks, I've released v2.0.4.
